x-pack/plugins/security_solution/server/endpoint/services/metadata/endpoint_metadata_service.ts

```diff
@@ -442,23 +438,18 @@ export class EndpointMetadataService {
         const agentPolicy = agentPoliciesMap[_agent.policy_id!];
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
         const endpointPolicy = endpointPoliciesMap[_agent.policy_id!];
-        // add the agent status from the fleet runtime field to
-        // the agent object
+
+        const runtimeFields = {
+          status: doc?.fields?.status?.[0] as AgentStatus,
+          last_checkin: doc?.fields?.last_checkin?.[0] as string | undefined,
```

Comment:
I believe the typecasting for `last_checkin` is actually redundant. The field values are already `string | undefined`. We do need the type cast for `status` as they are a union of literals and not `string`

Reply:
you're right, it's redundant now, but not because `last_checkin` is string. 😬 

this is the type of `fields`: `fields?: Record<string, any>;`, so the value is `any`, that's why I needed to cast `last_checkin`

but now that `runtimeFields` is `Partial<typeof _agent>`, both type assertions are unnecessary, indeed. so, removed: 9f7e99a5e418358ea10804f87b41a824944e1768